Sources/Atoms/Core/StoreContext.swift

```diff
@@ -201,11 +201,27 @@ internal struct StoreContext {
     }
 
     @usableFromInline
-    func reset(_ atom: some Atom) {
+    @_disfavoredOverload
+    func reset<Node: Atom>(_ atom: Node) {
+        let override = lookupOverride(of: atom)
+        let key = AtomKey(atom, overrideScopeKey: override?.scopeKey)
+
+        if let cache = lookupCache(of: atom, for: key) {
+            let newCache = makeNewCache(of: atom, for: key, override: override)
+            update(atom: atom, for: key, value: newCache.value, cache: cache, order: .newValue)
+        }
+    }
+
+    @usableFromInline
+    func reset<Node: Resettable>(_ atom: Node) {
         let override = lookupOverride(of: atom)
         let key = AtomKey(atom, overrideScopeKey: override?.scopeKey)
+        let state = getState(of: atom, for: key)
 
         if let cache = lookupCache(of: atom, for: key) {
+            let context = AtomCurrentContext(store: self, coordinator: state.coordinator)
+            atom.reset(context: context)
```

Comment:
@ra1028 
The most important part for review, as there are multiple potential options.

Since reset itself is a void function without side-effects, but also might have them depending on what is being executed (sample logging or operations affecting the store & cache). 

This suggested implementation has following key characteristics (that can be discussed)

- It executes regardless of `override` status. This is because code should not care about if atom is overriden or not when it comes to reset, given that it might be expecting key side-effect handling or state change, and override (which is commonly utilized in tests) would break that contract and make environment not clean.

- It executes _before_ new cache generation and update. Im not convinced myself if this is a proper approach here, and would like to get your input. Reason I chose this way is simply because if reset logic has important data handling that depend on state of the system, they can be lost if atom is reset and all subscribers are notified)

Reply:
> It executes regardless of override status

It's not necessary to take into account the override status. It's kind of a compromise, but `Refreshable` also calls the atom's custom `refresh` regardless of the atom is overridden or not.

> It executes before new cache generation and update.

Rather, do not create make a new cache after calling the custom `reset`. `Ressetable` should allow atoms to have arbitrary "reset" behavior. If you reset the dependencies of the atom, it re-creates the atom value transitively, if not, then it simply calls arbitrary reset behavior.

Reply:
> It's kind of a compromise, but `Refreshable` also calls the atom's custom refresh regardless of the atom is overridden or not.

```swift
if let override {
            value = override.value(atom)
        }
        else {
            let context = AtomCurrentContext(store: self, coordinator: state.coordinator)
            value = await atom.refresh(context: context)
        }
```

What am I missing www

Reply:
However, come to think of it, `refresh` may also follow how `reset` works that calls the custom `reset` regardless it's overridden 🤔